### PR TITLE
[359] Add DiagramDescription tests

### PR DESCRIPTION
**Pull request by gdaniel**
Bug: https://github.com/eclipse-syson/syson/issues/359

This PR adds a new maven module: `syson-diagram-tests` with reusable components used to define DiagramDescription tests. I followed the organization Sirius Web has, for example the `sirius-components-diagrams-tests` module.

This PR adds several checks on DiagramDescriptions, and includes some fixes that were discovered while running the tests.

### CODE REVIEW

File: backend/application/syson-application/src/test/java/org/eclipse/syson/JavaServiceIsCalledChecker.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import org.eclipse.syson.sysml.helper.EMFUtils;
+
+/**
+ * Checks that a {@link Method} is called by at least one AQL expression in the given {@link View}.
+ *
+ * @author gdaniel
+ */
+public class JavaServiceIsCalledChecker {
+
+    private static final String AQL_PREFIX = "aql:";
```
Comment (AxelRICHARD) [inline backend/application/syson-application/src/test/java/org/eclipse/syson/JavaServiceIsCalledChecker.java:53]: Please check that you have access to `org.eclipse.syson.util.AQLConstants` form here. If so, please use `AQLConstants.AQL` instead.

File: backend/views/syson-diagram-actionflow-view/pom.xml [hunk 2]
```diff
@@ -116,6 +116,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.eclipse.syson</groupId>
+			<artifactId>syson-diagram-tests</artifactId>
+			<version>2024.5.0</version>
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-actionflow-view/pom.xml:122]: May be add `<scope>test</scope>`?

Comment (AxelRICHARD) [reply backend/views/syson-diagram-actionflow-view/pom.xml:122]: Same remark for others pom.xml

File: backend/application/syson-application/src/test/java/org/eclipse/syson/JavaServiceIsCalledChecker.java (hunk 1 above)
Comment (gdaniel) [reply backend/application/syson-application/src/test/java/org/eclipse/syson/JavaServiceIsCalledChecker.java]: done

File: backend/views/syson-diagram-actionflow-view/pom.xml (hunk 2 above)
Comment (gdaniel) [reply backend/views/syson-diagram-actionflow-view/pom.xml:122]: done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**